capstone/src/main/webapp/script.js

```diff
@@ -77,37 +77,71 @@ async function loadAnnouncements () {
   const color1 = "#AAA";
   const color2 = "#BBB";
   var evenOdd = true;
-  for (var announcement in json) {
-    console.log(json[announcement]);
+  for (var announcement of json) {
+    const id = announcement.author + announcement.content + announcement.time; //Unique string to identiy this announcement.
+
     template.content.querySelector('img').src = 'images/logo.png';
-    template.content.querySelector('#announcement-author').innerHTML = json[announcement].authorName;
-    template.content.querySelector('#announcement-content').innerHTML = json[announcement].content;
+    template.content.querySelector('.announcement-author').innerHTML = announcement.authorName;
+    template.content.querySelector('.announcement-content').innerHTML = announcement.content;
+    template.content.querySelector('.announcement-content').id = id;
+    
 
-    const dateString = new Date(json[announcement].time).toLocaleDateString("en-US");
-    const timeString = new Date(json[announcement].time).toLocaleTimeString("en-US");
-    template.content.querySelector('#announcement-time').innerHTML = timeString + " on " + dateString;
+    const dateString = new Date(announcement.time).toLocaleDateString("en-US");
+    const timeString = new Date(announcement.time).toLocaleTimeString("en-US");
+    template.content.querySelector('.announcement-time').innerHTML = timeString + " on " + dateString;
 
     backgroundColor = evenOdd ? color1 : color2; //In order to switch background colors every announcement
     template.content.querySelector('#announcement-container').style.backgroundColor = backgroundColor;
     evenOdd = !evenOdd;
 
-    console.log(json[announcement].isAuthor);
-    console.log(JSON.parse(json[announcement].isAuthor));
-    if (JSON.parse(json[announcement].isAuthor)) {
+    if (JSON.parse(announcement.isAuthor)) {
       template.content.querySelector('.delete-announcement').style = "display: inline-block;";
-      template.content.querySelector('#club').value = json[announcement].club;
-      template.content.querySelector('#author').value = json[announcement].author;
-      template.content.querySelector('#content').value = json[announcement].content;
-      template.content.querySelector('#time').value = json[announcement].time;
+      template.content.querySelector('.edit-announcement').style = "display: inline-block;";
```

Comment:
nit: could you use .style.display = "inline-block" instead? Same for rest of the file
https://www.w3schools.com/jsref/prop_style_display.asp